text/0000-more-promise-methods.md

```diff
@@ -0,0 +1,63 @@
+- Feature Name: more-promise-methods
+- Start Date: 2017-02-25
+- RFC PR: 
+- Pony Issue: 
+
+# Summary
+
+This RFC proposes adding additional methods to the Promise actor.
+
+# Motivation
+
+The Promise actor is a useful tool for representing a value before it is available, though the public API is a bit sparse. Only 3 basic behaviors and functions are currently available to the user (apply, reject, and next). The following additional methods will improve composability and reflect common operations found in other promises/futures libraries.
+
+# Detailed design
+
+```pony
+
+fun tag join[B: Any #share = A](p: Promise[B]): Promise[(A, B)]
```

Comment:
If this method were named `add`, it could take advantage of the `+` operator sugar, which seems nice to me.

Maybe this method could be named `add`, and the `many` method could become `join`?